tests/mir-opt/instcombine_duplicate_switch_targets_e2e.rs

```diff
@@ -0,0 +1,16 @@
+// compile-flags: -Zmir-opt-level=2 -Zinline-mir
+// ignore-debug: standard library debug assertions add a panic that breaks this optimization
+#![crate_type = "lib"]
```

Comment:
Unrelated: should we make mir-opt tests "lib" by default, to avoid to list all the cases in a `fn main` everywhere?

Reply:
If we can. The codegen tests could do with the same treatment; I pulled this style of testing from them. There's a `#![crate_type = "lib"]` in nearly every single codegen test file.